### PR TITLE
Remove ShadowConfiguration

### PR DESCRIPTION
**Pull request by jongerrish**
Seems like like we can use 100% frameworkcode here.

Only thing on the shadow is really the qualifiers, but they don't seem
to be used for anything in Robolectric.

Add a test to show that #2909 is not an issue.

### CODE REVIEW

File: robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowView.java [hunk 1]
```diff
@@ -133,7 +134,7 @@ protected Drawable buildDrawable(int resourceId) {
   }
 
   protected String getQualifiers() {
-    return shadowOf(realView.getResources().getConfiguration()).getQualifiers();
```
Comment (xian) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowView.java]: Maybe we want to keep this lookup happening without going through `RuntimeEnvironment` because injection over statics? Seems possible that views could have different origin Contexts and Resources, especially in a multi-app case?

File: robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowConfiguration.java [hunk 2]
```diff
@@ -14,153 +14,5 @@
...
-    realConfiguration.screenLayout = Configuration.SCREENLAYOUT_LONG_NO |
-        Configuration.SCREENLAYOUT_SIZE_NORMAL;
-  }
-
-  @Implementation
-  public void setLocale( Locale l ) {
-    realConfiguration.locale = l;
-  }
-
-  public void overrideQualifiers(String qualifiers) {
```
Comment (xian) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowConfiguration.java]: Is there a way we could `@Deprecate` this for one release?

File: robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowView.java (hunk 1 above)
Comment (jongerrish) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowView.java]: Configuration.getQualifiers() and ShadowView.getQualifiers() are not used in Robolectric and not used in our internal codebase either so I doubt they are used much if at all. I'm not sure why qualifiers are exposed on the view at all.

I agree with the sentiment about statics, I think RuntimeEnvironment.setQualifiers() is "supposed" to be a method used only internally in Robolectric and @Config(qualifiers = "...") is the correct way of doing things.

Configuration changes are handled by activities through Activity.onCongifurationChanged(Configuration) which the test writer would simulate through ActivityController.configurationChange().

We should probably be invoking activity.getResources().calcConfigChanges(configuration) in the controller also so that the underlying resources are changed so the user would actually get the landscape resources over the portrait if simulating a screen rotation for example.

Most users seem happy to create this object themselves and pass it in, but we could always have a helper method to construct a Configuration object from a qualifiers string. 

So long story short, the setQualifiers() on Configuration doesn't actually do anything nor seems to be used, so I think its vestigial code left over from way back when.....

File: robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowConfiguration.java (hunk 2 above)
Comment (jongerrish) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowConfiguration.java]: Yeah, we could to avoid breakages but calls to this method don't actually seem to do anything anyway. See previous comments about usage etc.

**Comment by xian** [comment]
LGTM :-)
